src/test/java/com/task/ui/tests/OrderBookTest.java

```diff
@@ -0,0 +1,72 @@
+package com.task.ui.tests;
+
+import com.task.api.dto.*;
+import com.task.common.TestData;
+import com.task.ui.pages.*;
+import io.restassured.http.Header;
+import org.apache.http.HttpStatus;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+public class OrderBookTest extends BaseUiTest {
+
+    @Test
+    @DisplayName("E2E UI test")
+    public void testOrderBook() {
+        HomePage homePage = new HomePage(driver);
+        List<BookDTO> books = homePage.findBooks("HP");
+        Assertions.assertEquals(7, books.size());
+        books.forEach(b -> Assertions.assertNotNull(b.getCoverFileName()));

```

Comment:
Test will be more accurate and stable if it searches book with unique name.
Searching for the list of elements should be moved to separate test.